### PR TITLE
Add computer player

### PR DESCRIPTION
**Pull request by lrharris215**
Added a Computer Player class to handle the AI. The Computer Player can generate random PlaceRequests and HitRequests, as well as update both boards with their moves. I also created a RequestFactory to simplify the creation of new Requests.

### CODE REVIEW

File: src/main/java/com/battleship/backend/models/ComputerPlayer.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    public Boardable getComputerBoard() {
+        return computerBoard;
+    }
+
+    public String getName() {
+        return name;
+    }
+
```
Comment (kudukacar) [inline src/main/java/com/battleship/backend/models/ComputerPlayer.java:112]: Why do have these getters?  Is it just for tests?  In which case, you don't need it.

Comment (kudukacar) [reply src/main/java/com/battleship/backend/models/ComputerPlayer.java:112]: Same comment for the setter.

File: src/main/java/com/battleship/backend/models/ComputerPlayer.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                j++;
+            }
+        }
+        setShipList(newShipList);
+    }
+
+    public void fire(Request hitRequest){
+        playerBoard.hitSection(hitRequest.getRow(), hitRequest.getCol());
+    }
+
```
Comment (kudukacar) [inline src/main/java/com/battleship/backend/models/ComputerPlayer.java:90]: I think it's odd and violates single responsibility principle to have `placeShip`, `placeAllShips`, `removeShip`, and `Fire` within the ComputerPlayer.  The ComputerPlayer seems bloated.  Think of the humanPlayer - or think of requests that come to the controller - you will just have a place request and a hit request.

File: src/main/java/com/battleship/backend/models/ComputerPlayer.java (hunk 1 above)
Comment (lrharris215) [reply src/main/java/com/battleship/backend/models/ComputerPlayer.java:112]: -Yeah those were basically for the tests. The setter is used though, in removeShip. Will remove the unnecessary ones.

File: src/main/java/com/battleship/backend/models/ComputerPlayer.java (hunk 2 above)
Comment (lrharris215) [reply src/main/java/com/battleship/backend/models/ComputerPlayer.java:90]: So I've been thinking about this for a while now, and I'm not sure where to move those functions to. Should I remove them altogether, and just let the future Game handle all the Requests?

Comment (lrharris215) [reply src/main/java/com/battleship/backend/models/ComputerPlayer.java:90]: Like, the Computer player generates their Requests, and the Human player will have the logic needed to receive input from the controllers, and then the Game can decide what to do with those?

Comment (kudukacar) [reply src/main/java/com/battleship/backend/models/ComputerPlayer.java:90]: I think a Game class makes a lot of sense

**Review by kudukacar — APPROVED** [review]
Looks good!

**Pull request merged**